08_multibundle_requirejs/Gruntfile.js

```diff
@@ -0,0 +1,286 @@
+
+                    // lib
+                    'app/lib/**/*.js',
+
+                    'app/views/home/**/*.js',
+                    'app/controllers/home_controller.js',
+                ],
+                // Creates separate bundle for user page components – `<destination>/user.<hash>.js`
+                user: [
+                    'app/models/users_bundle/**/*.js',
+                    'app/collections/users_bundle/**/*.js',
+                    'app/views/users_bundle/**/*.js',
+                    'app/controllers/users_bundle/**/*.js',
+                    'public/js/app/templates/users_bundle/compiledTemplates.js'
+                ],
+
+                // Creates separate bundle for repos page components – `<destination>/repos.<hash>.js`
+                repos: [
+                    'app/models/repos_bundle/**/*.js',
+                    'app/collections/repos_bundle/**/*.js',
+                    'app/views/repos_bundle/**/*.js',
+                    'app/controllers/repos_bundle/**/*.js',
+                    'public/js/app/templates/repos_bundle/compiledTemplates.js'
+                ]
+            }
+        },
+
+        handlebars: {
+            compile_server: {
+                options: {
+                    namespace: false,
+                    commonjs: true,
+                    processName: function(filename) {
+                        return filename.replace('app/templates/', '').replace('.hbs', '');
+                    }
+                },
+                src: 'app/templates/**/*.hbs',
+                dest: 'app/templates/compiledTemplates.js',
+                filter: function(filepath) {
+                    var filename = path.basename(filepath);
+                    // Exclude files that begin with '__' from being sent to the client,
+                    // i.e. __layout.hbs.
+                    return filename.slice(0, 2) !== '__';
+                }
+            },
+            /*
+             * Create separate compiled templates so that they can be bundled
+             */
+            compile_client: {
+                options: {
+                    amd: true,
+                    processName: function(filename) {
+                        return filename.replace('app/templates/', '').replace('.hbs', '');
+                    }
+                },
+                files: {
+                    'public/js/app/templates/compiledTemplates.js': ['app/templates/**/*.hbs',
+                        '!' + usersBundleTemplatePath, // Exclude since we'll build it separate
+                        '!' + reposBundleTemplatePath, // Exclude since we'll build it separate
+                        '!app/templates/__layout.hbs'
+                    ],
+                    'public/js/app/templates/users_bundle/compiledTemplates.js': usersBundleTemplatePath,
+                    'public/js/app/templates/repos_bundle/compiledTemplates.js': reposBundleTemplatePath
+                }
```

Comment:
Thinking more over it, I think I got the same feeling, that incorporating bundles into app structure makes it less (like in "no") flexible. And developers need to know what bundles have what modules.

It would definitely give better exposure to bundles themselves and reduce "cross pollination". Which will be good for performance, but maybe not as good codebase evolution.

It feels like old argument Folders vs. Tags in email universe.

As one of the examples where our current structure (bundles specified only in the confit) helped us is when ad blockers were introduced in iOS9 and we had to pull some of the "hot" files into "optional" bundle that we expect not to load in some cases.

Having bundle name as part of the module id (require path) would make it harder and more confusing.

On other hand it might be easier to grasp when you just starting to understand the whole concept.

Bottom line we need more examples. :)
And thanks to @crwang for breaking ground on it. :)